### ISSUE TITLE
Apple Pay returns Your payment information is formatted improperly.

### ISSUE THREAD

**Issue by ordinaryman09** [tipsi/tipsi-stripe#241]
Hi,

I'm using the 4.6.2 version and I'm getting  Your payment information is formatted improperly. Please make sure you're correctly using the latest version of our iOS library. For more information see when calling the API.

`const items = [{
  label: 'Whisky',
  amount: '50.00',
}, {
  label: 'Tipsi, Inc',
  amount: '50.00',
}]

const options = {
  
}

try {
const token = yield stripe.paymentRequestWithApplePay(items, options)
console.log("token is " + token); 
stripe.completeApplePayRequest()
}
catch (error) {
  console.log("error is " + error);
}`

**Comment by isnifer** [comment]
@ordinaryman09 Hi! What version of Pod do you use?

**Comment by ordinaryman09** [comment]
I'm using Stripe version 11.0.0

**Comment by shokimble** [comment]
@ordinaryman09 remove the quotes from amount (just looking at my code there's a comment saying it needs to be a string but I've got toFixed(2) so it's a float). If that doesn't sort it I also added a total item to items:

{
label: 'Pay',
amount: 100.00,
}

Also from memory it's fussy about options so try adding this:

		let options = {
			requiredShippingAddressFields: ['name','email','postal_address'],
			currencyCode: 'USD',
			//countryCode: 'US',
			
			'shippingMethods': [
				{
					id: 'free',
					label: 'Free',
					detail: 'Free Shipping',
					amount: '0'
				}
			]
			
		}

**Comment by tmaly1980** [comment]
@shokimble, you mean parseFloat(amount) ? I'm getting this error, so I think it NEEDS to be a string:

Error: Failed items type: Invalid items `items[0].amount` of type `number` supplied to `Stripe.paymentRequestWithApplePay`, expected `string`.

**Comment by shokimble** [comment]
Sorry yes re-read what I said. Not sure what I was thinking.

toFixed generates a string.

Might want to post here what you have. It's very fiddly but i can assure you it works(at least the version I'm using). I have an app on the store with Apple pay.

**Comment by tmaly1980** [comment]
This is what I have:
```
      stripe_token = yield stripe.paymentRequestWithApplePay([
        { label: 'Purchase', amount: "900.00" },
        { label: 'Company, Inc', amount: "900.00" },
      ], {
        requiredBillingAddressFields: ['all']
      })
```

**Comment by kyte0017** [comment]
@tmaly1980 Have you had any luck?

I am getting the same error as you when I am running it on the simulator. When I run it on a physical device, the app still crashes, but I noticed I get a success request on my stripe dashboard. So I think it has something to do with apple pay information format...

**Comment by tmaly1980** [comment]
This still is unresolved, I've looked through the dashboard and I get the invalid_request_error . One thing that might play a part is making sure that the apple pay certificate gets uploaded back onto Stripe under Payments -> Apple Pay. I didn't have that. But it's still not working and now I'm going to try to upgrade my native Stripe library version as well as the API version, since that seems to be dated back to 2015-10-12 . 

I've also emailed stripe and they say they'll get back within 24 hours, I'll definitely post results here as soon as I know.

**Comment by yegorsch** [comment]
@tmaly1980 Did you find a way to solve it?

**Comment by tmaly1980** [comment]
Yes, I figured it out. The issue I was having was that I never uploaded the apple pay certificate back onto stripe's dashboard. The Stripe Apple Pay docs explain how. Did you follow them, including getting the finished certificate uploaded back onto Stripe? This error message isn't helpful with figuring that out. 

After I resolved this issue, I wrote the code in such a way that the Apple Pay token received back was treated as if it were a credit card / payment source to be added to the customer's account. I don't know yet how Apple is going to respond to this approach. My impression is that Apple Pay is a one-time authorization for a specific amount, but Stripe treats it like a reusable payment source, as if it were a credit card. I've been able to add the apple pay token into my cards and initiate charges using it as the default source, without prompting the user every time. It works, but for all I know, Apple may reject the app.
